Specs/Core/EllipsoidRhumbLineSpec.js

```diff
@@ -0,0 +1,581 @@
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+        var distance = Math.cos(fortyfiveDegrees) * CesiumMath.PI_OVER_TWO * 6;
+        expect(distance).toEqualEpsilon(rhumb.surfaceDistance, CesiumMath.EPSILON12);
+    });
+
+    it('tests sphere', function() {
+        var radius = 6378137.0;
+        var ellipsoid = new Ellipsoid(radius, radius, radius);
+        var fifteenDegrees = Math.PI / 12;
+        var initial = new Cartographic(fifteenDegrees, fifteenDegrees);
+        var distance = radius * fifteenDegrees;
+
+        var rhumb1 = EllipsoidRhumbLine.fromStartHeadingDistance(initial, fifteenDegrees, distance, ellipsoid);
+        var rhumb2 = new EllipsoidRhumbLine(initial, rhumb1.end, ellipsoid);
+
+        expect(fifteenDegrees).toEqualEpsilon(rhumb2.heading, CesiumMath.EPSILON12);
+        expect(distance).toEqualEpsilon(rhumb2.surfaceDistance, CesiumMath.EPSILON6);
+    });
+
+    it('tests sphereoid', function() {
+        var ellipsoid = Ellipsoid.WGS84;
+        var fifteenDegrees = Math.PI / 12;
+        var initial = new Cartographic(fifteenDegrees, fifteenDegrees);
+        var distance = ellipsoid.maximumRadius * fifteenDegrees;
+
+        var rhumb1 = EllipsoidRhumbLine.fromStartHeadingDistance(initial, fifteenDegrees, distance, ellipsoid);
+        var rhumb2 = new EllipsoidRhumbLine(initial, rhumb1.end, ellipsoid);
+
+        expect(fifteenDegrees).toEqualEpsilon(rhumb2.heading, CesiumMath.EPSILON12);
+        expect(distance).toEqualEpsilon(rhumb2.surfaceDistance, CesiumMath.EPSILON6);
+    });
+
+    it('tests sphere close to 90 degrees', function() {
+        var radius = 6378137.0;
+        var ellipsoid = new Ellipsoid(radius, radius, radius);
+        var fifteenDegrees = Math.PI / 12;
+        var initial = new Cartographic(fifteenDegrees, fifteenDegrees);
+        var distance = radius * fifteenDegrees;
+
+        var oneDegree = CesiumMath.RADIANS_PER_DEGREE;
+        var eightyNineDegrees = 89 * oneDegree;
+        var eightyNinePointNineDegrees = 89.9 * oneDegree;
+        var ninetyDegrees = 90 * oneDegree;
+        var ninetyPointOneDegrees = 90.1 * oneDegree;
+        var ninetyPointZeroTwoDegrees = 90.02 * oneDegree;
+
+        var rhumb1 = EllipsoidRhumbLine.fromStartHeadingDistance(initial, eightyNineDegrees, distance, ellipsoid);
+        var rhumb2 = new EllipsoidRhumbLine(initial, rhumb1.end, ellipsoid);
+        expect(rhumb1.heading).toEqualEpsilon(rhumb2.heading, CesiumMath.EPSILON12);
+        expect(rhumb1.surfaceDistance).toEqualEpsilon(rhumb2.surfaceDistance, CesiumMath.EPSILON6);
+
+        rhumb1 = EllipsoidRhumbLine.fromStartHeadingDistance(initial, eightyNinePointNineDegrees, distance, ellipsoid);
+        rhumb2 = new EllipsoidRhumbLine(initial, rhumb1.end, ellipsoid);
+        expect(rhumb1.heading).toEqualEpsilon(rhumb2.heading, CesiumMath.EPSILON12);
+        expect(rhumb1.surfaceDistance).toEqualEpsilon(rhumb2.surfaceDistance, CesiumMath.EPSILON6);
```

Comment:
Should this also check that the distance increases as the point approaches the pole? Looks to me like this spec is intended to demonstrate how rhumbs spiral.

Reply:
I don't think it's necessary, it more designed to ensure that the 90 degree code paths are executed and working correctly.